pkg/kritis/crd/securitypolicy/securitypolicy_test.go

```diff
@@ -75,14 +76,12 @@ func Test_UnqualifiedImage(t *testing.T) {
 		},
 	}
 	violations, err := ValidateImageSecurityPolicy(isp, "", &testutil.MockMetadataClient{})
-	expected := []Violation{
-		{
-			Vulnerability: metadata.Vulnerability{},
-			Violation:     UnqualifiedImageViolation,
-			Reason:        UnqualifiedImageReason(""),
-		},
+	testutil.CheckError(t, false, err)
+	if len(violations) != 1 {
+		t.Errorf("expected only 1 violation. Got %d", len(violations))
 	}
-	testutil.CheckErrorAndDeepEqual(t, false, err, violations, expected)
+	testutil.CheckErrorAndDeepEqual(t, false, err, violations[0].Type(), policy.UnqualifiedImageViolation)
```

Comment:
Would it be possible to compare the result against against expectations and showing a diff for all the fields that don't match? It's generally recommended in Go as a way to make tests easier to reason about and maintain.

Context: http://go/go-test-comments#compare-full-structures and http://go/go-test-comments#print-diffs